rest/src/main/java/io/syndesis/rest/v1/handler/user/UserHandler.java

```diff
@@ -15,30 +15,49 @@
  */
 package io.syndesis.rest.v1.handler.user;
 
+import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.xml.ws.Response;
 
 import io.syndesis.dao.manager.DataManager;
 import io.syndesis.model.Kind;
 import io.syndesis.model.user.User;
+import io.syndesis.openshift.OpenShiftService;
 import io.syndesis.rest.v1.handler.BaseHandler;
 import io.syndesis.rest.v1.operations.Getter;
 import io.syndesis.rest.v1.operations.Lister;
 import io.swagger.annotations.Api;
 
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
 
 @Path("/users")
 @Api(value = "users")
 @Component
 public class UserHandler extends BaseHandler implements Lister<User>, Getter<User> {
 
-    public UserHandler(DataManager dataMgr) {
+    private OpenShiftService openShiftService;
+
+    public UserHandler(DataManager dataMgr, OpenShiftService openShiftService) {
         super(dataMgr);
+        this.openShiftService = openShiftService;
     }
 
-     @Override
+    @Override
     public Kind resourceKind() {
         return Kind.User;
     }
-}
 
+    @Path("~")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public User whoAmI() {
+        String token = String.valueOf(SecurityContextHolder.getContext().getAuthentication().getCredentials());
```

Comment:
Doesn't the authentication subject contains something like a user name, too ? (I ask because we use that it at some other place).

Reply:
there is a [getPrincipal()](https://docs.spring.io/spring-security/site/docs/current/apidocs/org/springframework/security/core/Authentication.html#getPrincipal--) method that should return the username

Reply:
This returns a User object containing extra info, eg full name, so we need to make a call to auth source. In time, we can expand this to include roles, etc if required.

Reply:
ok, just want to get sure that getPrincipal() also works at another place

Reply:
Yes it does. This is only to retrieve further info about the user that is not available in the oauth callback/headers.